core/src/main/java/io/questdb/cutlass/http/HttpConnectionContext.java

```diff
@@ -926,6 +955,11 @@ private boolean handleClientRecv(HttpRequestProcessorSelector selector, Reschedu
                     } catch (CairoException e) {
                         processor = rejectRequest(HTTP_FORBIDDEN, e.getFlyweightMessage());
                     }
+
+                    if (!connectionCounted) {
```

Comment:
there seem to be a flaw there. Connection counter is never revalidated irrespective of success or failure. So a scenario possible where connection is rejected initially the second attempt will go thru because context is no re-validated

Reply:
there might be a test missing that does the above